src/objective-c/tests/InteropTests.m

```diff
@@ -151,6 +152,230 @@ - (dispatch_queue_t)dispatchQueue {
+             finishHook:(void (^)(GRPCInterceptorManager *manager))finishHook
+receiveNextMessagesHook:(void (^)(NSUInteger numberOfMessages,
+                                  GRPCInterceptorManager *manager))receiveNextMessagesHook
+     responseHeaderHook:(void (^)(NSDictionary *initialMetadata,
+                                  GRPCInterceptorManager *manager))responseHeaderHook
+       responseDataHook:(void (^)(id data, GRPCInterceptorManager *manager))responseDataHook
+      responseCloseHook:(void (^)(NSDictionary *trailingMetadata, NSError *error,
+                                  GRPCInterceptorManager *manager))responseCloseHook
+       didWriteDataHook:(void (^)(GRPCInterceptorManager *manager))didWriteDataHook {
+  if ((self = [super init])) {
+    _dispatchQueue = dispatchQueue;
+    _startHook = startHook;
+    _writeDataHook = writeDataHook;
+    _finishHook = finishHook;
+    _receiveNextMessagesHook = receiveNextMessagesHook;
+    _responseHeaderHook = responseHeaderHook;
+    _responseDataHook = responseDataHook;
+    _responseCloseHook = responseCloseHook;
+    _didWriteDataHook = didWriteDataHook;
+    _dispatchQueue = dispatchQueue;
+  }
+  return self;
+}
+
+- (GRPCInterceptor *)createInterceptorWithManager:(GRPCInterceptorManager *)interceptorManager {
+  return [[HookIntercetpor alloc] initWithInterceptorManager:interceptorManager
+                                               dispatchQueue:_dispatchQueue
+                                                   startHook:_startHook
+                                               writeDataHook:_writeDataHook
+                                                  finishHook:_finishHook
+                                     receiveNextMessagesHook:_receiveNextMessagesHook
+                                          responseHeaderHook:_responseHeaderHook
+                                            responseDataHook:_responseDataHook
+                                           responseCloseHook:_responseCloseHook
+                                            didWriteDataHook:_didWriteDataHook];
+}
+
+@end
+
+@implementation HookIntercetpor {
+  void (^_startHook)(GRPCRequestOptions *requestOptions, GRPCCallOptions *callOptions,
+                     GRPCInterceptorManager *manager);
+  void (^_writeDataHook)(id data, GRPCInterceptorManager *manager);
+  void (^_finishHook)(GRPCInterceptorManager *manager);
+  void (^_receiveNextMessagesHook)(NSUInteger numberOfMessages, GRPCInterceptorManager *manager);
+  void (^_responseHeaderHook)(NSDictionary *initialMetadata, GRPCInterceptorManager *manager);
+  void (^_responseDataHook)(id data, GRPCInterceptorManager *manager);
+  void (^_responseCloseHook)(NSDictionary *trailingMetadata, NSError *error,
+                             GRPCInterceptorManager *manager);
+  void (^_didWriteDataHook)(GRPCInterceptorManager *manager);
+  GRPCInterceptorManager *_manager;
+  dispatch_queue_t _dispatchQueue;
```

Comment:
All of our examples, tests use a single dispatch queue for requests and responses, which means that requests and responses won't be processed in parallel. Can we add a test where we use different queues for requests and responses so that we have test coverage for concurrent requests, responses?